src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -139,6 +142,46 @@ class ObservationStore(
         }
   }
 
+  fun fetchStartableObservations(
+      plantingSiteId: PlantingSiteId? = null
+  ): List<ExistingObservationModel> {
+    val maxStartDate = LocalDate.ofInstant(clock.instant(), ZoneOffset.UTC).plusDays(1)
+    val timeZoneField =
+        DSL.coalesce(
+            OBSERVATIONS.plantingSites.TIME_ZONE,
+            OBSERVATIONS.plantingSites.organizations.TIME_ZONE)
+
+    return dslContext
+        .select(OBSERVATIONS.asterisk(), timeZoneField)
+        .from(OBSERVATIONS)
+        .where(OBSERVATIONS.STATE_ID.eq(ObservationState.Upcoming))
+        .and(OBSERVATIONS.START_DATE.le(maxStartDate))
+        .andExists(
+            // When we test whether a subzone is planted, we need to account for reassignments
+            // by totaling the number of plants in the plantings in the subzone, so we don't count
+            // a subzone as planted if all its deliveries were reassigned. But here, it is
+            // sufficient to just check for the existence of any planting; reassignments can only
+            // move plants between subzones within a single planting site, which means there's no
+            // way for a reassignment to lower a site's plant count to zero.
+            DSL.selectOne()
+                .from(PLANTINGS)
+                .where(PLANTINGS.PLANTING_SITE_ID.eq(OBSERVATIONS.PLANTING_SITE_ID))
+                .and(PLANTINGS.PLANTING_SUBZONE_ID.isNotNull))
+        .apply { if (plantingSiteId != null) and(OBSERVATIONS.PLANTING_SITE_ID.eq(plantingSiteId)) }
```

Comment:
Should this be before 159? (guessing the query is optimized anyways, maybe not an issue)

Reply:
Right, the order of conditionals isn't important. The query optimizer will use a bunch of heuristics to (hopefully) figure out an efficient execution plan that satisfies the constraints.